library/alloc/src/collections/mod.rs

```diff
@@ -58,7 +58,31 @@ use core::fmt::Display;
 /// The error type for `try_reserve` methods.
 #[derive(Clone, PartialEq, Eq, Debug)]
 #[unstable(feature = "try_reserve", reason = "new API", issue = "48043")]
-pub enum TryReserveError {
+pub struct TryReserveError {
+    detail: TryReserveErrorDetail,
+}
+
+impl TryReserveError {
+    /// Details about the allocation that caused the error
+    #[inline]
+    #[unstable(
+        feature = "try_reserve_detail",
+        reason = "Uncertain how much info should be exposed",
+        issue = "48043"
+    )]
+    pub fn detail(&self) -> TryReserveErrorDetail {
```

Comment:
i think this is usually called "kind" (like [`std::io::Error::kind()`](https://doc.rust-lang.org/std/io/struct.Error.html#method.kind)).

Reply:
I have considered that, but `ErrorKind` is a field-less enum. Besides, I think this type could be removed entirely in favor of getters on the error itself, or just dropped completely to make the error type small and efficient.

Reply:
Ok, I see there are more variations in `kind` methods in libstd, so I've renamed it anyway